awx/main/tests/functional/api/test_job_runtime_params.py

```diff
@@ -270,106 +270,33 @@ def test_job_launch_JT_with_validation(machine_credential, deploy_jobtemplate):
     final_job_extra_vars = yaml.load(job_obj.extra_vars)
     assert 'job_template_var' in final_job_extra_vars
     assert 'job_launch_var' in final_job_extra_vars
-    assert job_obj.credential.id == machine_credential.id
+    assert [cred.pk for cred in job_obj.credentials.all()] == [machine_credential.id]
 
 
 @pytest.mark.django_db
-@pytest.mark.parametrize('pks, error_msg', [
-    ([1], 'must be network or cloud'),
-    ([999], 'object does not exist'),
-])
-def test_job_launch_JT_with_invalid_extra_credentials(machine_credential, deploy_jobtemplate, pks, error_msg):
+def test_job_launch_with_default_creds(machine_credential, vault_credential, deploy_jobtemplate):
     deploy_jobtemplate.ask_credential_on_launch = True
-    deploy_jobtemplate.save()
-
-    kv = dict(extra_credentials=pks, credential=machine_credential.id)
+    deploy_jobtemplate.credentials.add(machine_credential)
+    deploy_jobtemplate.credentials.add(vault_credential)
+    kv = dict()
     serializer = JobLaunchSerializer(
         instance=deploy_jobtemplate, data=kv,
         context={'obj': deploy_jobtemplate, 'data': kv, 'passwords': {}})
     validated = serializer.is_valid()
-    assert validated is False
-
-
-@pytest.mark.django_db
-def test_job_launch_JT_enforces_unique_extra_credential_kinds(machine_credential, credentialtype_aws, deploy_jobtemplate):
-    """
-    JT launching should require that extra_credentials have distinct CredentialTypes
-    """
-    pks = []
-    for i in range(2):
-        aws = Credential.objects.create(
-            name='cred-%d' % i,
-            credential_type=credentialtype_aws,
-            inputs={
-                'username': 'test_user',
-                'password': 'pas4word'
-            }
-        )
-        aws.save()
-        pks.append(aws.pk)
-
-    kv = dict(extra_credentials=pks, credential=machine_credential.id)
-    serializer = JobLaunchSerializer(
-        instance=deploy_jobtemplate, data=kv,
-        context={'obj': deploy_jobtemplate, 'data': kv, 'passwords': {}})
-    validated = serializer.is_valid()
-    assert validated is False
-
-
-@pytest.mark.django_db
-@pytest.mark.parametrize('ask_credential_on_launch', [True, False])
-def test_job_launch_with_no_credentials(deploy_jobtemplate, ask_credential_on_launch):
```

Comment:
These next few tests fall under the "which credential(s) required to launch", which is no longer a thing.